### ISSUE TITLE
Add accounts to AccountSerialize and AccountDeserialize traits

### ISSUE THREAD

**Issue by armaniferrante** [project-serum/anchor#2]
As discussed with @sconybeare, we should explore adding an account deserialization trait that looks something like

```rust
impl TRAIT for STRUCT {
  fn deserialize(deserializer: D, accounts: &mut &[AccountInfo]) -> Self {
    ...
  }
}
```

**Comment by armaniferrante** [comment]
Preliminary work on this will be done under https://github.com/project-serum/anchor/issues/10. The work under this issue can extend the `AccountSerialize` and `AccountDeserialize` traits proposed there, to include an additional `accounts: &mut &[AccountInfo]` argument.

**Comment by armaniferrante** [comment]
Addressed by #21.
